### PR TITLE
Add copy shot to clipboard

### LINKED ISSUE #4776: [FxA] Add "Copy" to Shot Page
https://mozilla.github.io/testpilot-assets/Firefox_Screenshots/NEW_FxA_Integration/shot shows a "Copy" button.  This feature does not exist currently.

Is it copy URL or copy image?

### PR DESCRIPTION
**Pull request by chenba**
Fixes #4776

### CODE REVIEW

File: server/src/pages/shot/controller.js [hunk 1]
```diff
@@ -281,4 +283,14 @@ function render() {
   page.render(model);
 }
 
+document.addEventListener("addon-present", (e) => {
+  if (e.detail) {
+    const capabilities = JSON.parse(e.detail);
+    if (capabilities["copy-to-clipboard"]) {
```
Comment (chenba) [inline server/src/pages/shot/controller.js]: This prevents the copy button from being displayed to older versions of the addon.

File: server/src/pages/shot/view.js [hunk 2]
```diff
@@ -43,7 +43,11 @@ class Clip extends React.Component {
     if (!isValidClipImageUrl(clip.image.url)) {
       return null;
     }
-    const node = <img id="clipImage" style={{height: "auto", width: Math.floor(clip.image.dimensions.x) + "px", maxWidth: "100%" }} ref={clipImage => this.clipImage = clipImage} src={ clip.image.url } alt={ clip.image.text } />;
+    const node = <img id="clipImage"
+      style={{ height: "auto", width: Math.floor(clip.image.dimensions.x) + "px", maxWidth: "100%" }}
+      ref={clipImage => this.clipImage = clipImage}
+      src={clip.image.url}
+      alt={clip.image.text} />;
```
Comment (chenba) [inline server/src/pages/shot/view.js]: Only formatting changes.

File: server/src/pages/shot/view.js [hunk 3]
```diff
@@ -223,7 +231,9 @@ class Body extends React.Component {
     const clipNames = shot.clipNames();
     const clip = shot.getClip(clipNames[0]);
     return <reactruntime.BodyTemplate {...this.props}>
-        <Editor clip={clip} pngToJpegCutoff={this.props.pngToJpegCutoff} onCancelEdit={this.onCancelEdit.bind(this)} onClickSave={this.onClickSave.bind(this)}></Editor>
+        <Editor clip={clip} pngToJpegCutoff={this.props.pngToJpegCutoff}
+          onCancelEdit={this.onCancelEdit.bind(this)}
+          onClickSave={this.onClickSave.bind(this)}></Editor>
```
Comment (chenba) [inline server/src/pages/shot/view.js]: Formatting changes.

File: server/src/pages/shot/view.js [hunk 4]
```diff
@@ -339,26 +349,46 @@ class Body extends React.Component {
     let favoriteShotButton = null;
     let trashOrFlagButton = null;
     let editButton = null;
-    const highlight = this.state.highlightEditButton ? <div className="edit-highlight" onClick={ this.onClickEdit.bind(this) } onMouseOver={ this.onMouseOverHighlight.bind(this) } onMouseOut={ this.onMouseOutHighlight.bind(this) }></div> : null;
+    const highlight = this.state.highlightEditButton
+      ? <div className="edit-highlight"
+          onClick={this.onClickEdit.bind(this)}
+          onMouseOver={this.onMouseOverHighlight.bind(this)}
+          onMouseOut={this.onMouseOutHighlight.bind(this)}></div>
+      : null;
```
Comment (chenba) [inline server/src/pages/shot/view.js]: Formatting changes.

File: server/src/pages/shot/view.js [hunk 5]
```diff
@@ -339,26 +349,46 @@ class Body extends React.Component {
...
+
+    const downloadButton = <div className="download-shot-button">
+      <Localized id="shotPageDownloadShot">
+        <button className="nav-button icon-download" onClick={this.onClickDownload.bind(this)}
+          title="Download the shot image">
+          <Localized id="shotPageDownload">
+            <span className="btn-text">Download</span>
+          </Localized>
+        </button>
+      </Localized></div>;
```
Comment (chenba) [inline server/src/pages/shot/view.js]: Formatting changes. ☝️ 

New button. 👇

File: server/src/pages/shot/view.js [hunk 6]
```diff
@@ -339,26 +349,46 @@ class Body extends React.Component {
...
+
+    const downloadButton = <div className="download-shot-button">
+      <Localized id="shotPageDownloadShot">
+        <button className="nav-button icon-download" onClick={this.onClickDownload.bind(this)}
+          title="Download the shot image">
+          <Localized id="shotPageDownload">
+            <span>Download</span>
+          </Localized>
+        </button>
+      </Localized></div>;
```
Comment (chenba) [inline server/src/pages/shot/view.js]: Formatting changes. ☝️ 

New button. 👇

File: static/css/partials/_buttons.scss [hunk 7]
```diff
@@ -138,15 +138,4 @@
...
-  &.flag {
-    background-image: url("../img/icon-flag.svg");
-    &:hover {
-      background-color: $light-hover;
-    }
-
-    &:active {
-      background-color: $light-active;
-    }
-  }
```
Comment (chenba) [inline static/css/partials/_buttons.scss]: Leftover styles from the report shot flag that was moved to the footer in #4687.  The background image file is also removed in this PR.

**Comment by 6a68** [comment]
@chenba Looks like this needs a rebase

**Comment by chenba** [comment]
> We're also showing website copy button click errors as desktop/Firefox notifications, which is odd

+1 

And good point about right click to copy.

**Comment by 6a68** [comment]
@chenba Yeah, sorry I didn't think to raise these points before you did all the work in this PR :-P

**Comment by 6a68** [comment]
Talked about this in the morning meeting, and @johngruen wants to land it and see if people like it. Merging :+1:

File: server/src/pages/shot/view.js [hunk 8]
```diff
@@ -343,26 +356,47 @@ class Body extends React.Component {
...
+
+    const copyButton = <div className="copy-img-button" hidden={!this.state.canCopy}>
+      <Localized id="shotPageCopyButton">
+        <button className="nav-button icon-copy transparent copy"
+          title="Copy image to clipboard"
```
Comment (flodolo) [inline server/src/pages/shot/view.js]: Looks like this string wasn't exposed for localization. We'll extract is in #4935

**Pull request merged**